leptos_macro/src/lib.rs

```diff
@@ -608,6 +610,8 @@ pub fn component(args: proc_macro::TokenStream, s: TokenStream) -> TokenStream {
     let Ok(mut dummy) = syn::parse::<DummyModel>(s.clone()) else {
         return s;
     };
+    convert_impl_trait_to_generic(&mut dummy.sig);
+
```

Comment:
I guess this needs to be included in islands as well, or moved to `Parse for DummyModel`

Reply:
I'll just let the Dummy keep the original `impl Trait` since it doesn't actually need the conversion. Unsure how different the error messages would be with it